src/bin/ion/commands/beta/generate/generator.rs

```diff
@@ -126,31 +165,99 @@ impl<'a> CodeGenerator<'a> {
         // add imports for the template
         context.insert("imports", &imports);
 
-        // generate read and write APIs for the data model
-        self.generate_read_api(&mut context, &mut tera_fields, &mut code_gen_context)?;
-        self.generate_write_api(&mut context, &mut tera_fields, &mut code_gen_context);
-        modules.push(self.language.file_name(&data_model_name));
+        // add fields for template
+        if code_gen_context.abstract_data_type == Some(AbstractDataType::Struct)
+            || code_gen_context.abstract_data_type == Some(AbstractDataType::Value)
+            || matches!(
+                code_gen_context.abstract_data_type,
+                Some(AbstractDataType::Sequence(_))
+            )
+        {
+            context.insert("fields", &tera_fields);
+            if let Some(abstract_data_type) = &code_gen_context.abstract_data_type {
+                context.insert("abstract_data_type", abstract_data_type);
+            } else {
+                return invalid_abstract_data_type_error(
+                    "Can not determine abstract data type, constraints are mapping not mapping to an abstract data type.",
+                );
+            }
+        }
+
+        self.render_generated_code(
+            modules,
+            &abstract_data_type_name,
+            &mut context,
+            &mut code_gen_context,
+        )
+    }
+
+    fn render_generated_code(
```

Comment:
(🗺️ PR tour) This method is just extracted from previous `generate` method for better modularization of the code.